vcf-dumper/vcf-dumper-lib/src/main/java/uk/ac/ebi/eva/vcfdump/VariantToVariantContextConverter.java

```diff
@@ -86,57 +96,61 @@ public VariantContext transform(Variant variant) {
-                        .append(symbol != null ? symbol : "").append("|")
-                        .append(gene != null ? gene : "").append("|")
-                        .append(feature != null ? feature : "").append("|")
-                        .append(bioType != null ? bioType : "").append("|")
-                        .append(cDnaPosition != null ? cDnaPosition : "").append("|")
-                        .append(cdsPosition != null ? cdsPosition : "");
-                consequences.add(csqSb.toString());
-            }
-            csq = String.join(",", consequences);
-
+            String allele = variant.getAlternate();
+            csq = consequenceTypes.stream()
+                    .map(consequenceType -> transformConsequenceTypeToCsqTag(allele, consequenceType))
+                    .collect(Collectors.joining(","));
         }
         return csq;
     }
 
-    private String[] getAllelesArray(Variant variant) {
-        String[] allelesArray;
-        // if there are indels, we cannot use the normalized alleles (hts forbids empty alleles), so we have to extract a context allele
-        // from the VCF source line, add it to the variant and update the variant coordinates
-        if (variant.getReference().isEmpty() || variant.getAlternate().isEmpty()) {
-            variant = updateVariantAddingContextNucleotideFromSourceLine(variant);
+    private String transformConsequenceTypeToCsqTag(String allele, ConsequenceType consequenceType) {
+        Set<Integer> soAccessions =  consequenceType.getSoAccessions();
+        String soNames = null;
+        if (soAccessions != null) {
+            soNames = soAccessions.stream()
+                    .map(ConsequenceTypeMappings::getSoName).filter(Objects::nonNull)
+                    .collect(Collectors.joining("&"));
         }
+        String symbol = consequenceType.getGeneName();
+        String gene = consequenceType.getEnsemblGeneId();
+        String feature = consequenceType.getEnsemblTranscriptId();
+        String bioType = consequenceType.getBiotype();
+        Integer cDnaPosition = consequenceType.getcDnaPosition();
+        Integer cdsPosition = consequenceType.getCdsPosition();
+
+        StringBuilder csqSb = new StringBuilder();
+        csqSb.append(allele != null ? allele : "").append("|")
+                .append(soNames != null ? soNames : "").append("|")
+                .append(symbol != null ? symbol : "").append("|")
+                .append(gene != null ? gene : "").append("|")
+                .append(feature != null ? feature : "").append("|")
+                .append(bioType != null ? bioType : "").append("|")
+                .append(cDnaPosition != null ? cDnaPosition : "").append("|")
+                .append(cdsPosition != null ? cdsPosition : "");
+        return csqSb.toString();
+    }
+
+    private String[] getAllelesArray(VariantWithSamplesAndAnnotation variant) {
+        String[] allelesArray;
```

Comment:
This line and the next can be merged in a single one:
` String[] allelesArray = new String[] ....`